src/analysis/retail/evoker/devastation/modules/talents/Causality.tsx

```diff
@@ -53,165 +64,164 @@ class Causality extends Analyzer {
       Events.damage.by(SELECTED_PLAYER).spell(SPELLS.PYRE),
       this.pyreReduceCooldown,
     );
-
-    this.addEventListener(
-      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS.DRAGONRAGE_TALENT),
-      this.dragonRage,
-    );
   }
 
-  // Pyre can at most trigger 5 CDR events per cast (this includes Pyres procced from Volatility and Dragonrage)
-  // With how dragonrage pyres work however, that would actually be 15 events (3xpyres)
+  /** Pyre triggers CDR events based on amount of targets hit, up to a maximum of 5.
+   * Pyres have a traveltime and therefore sometimes we have some funky behaviour
+   * eg. Dragonrage casts out 3 Pyres, at 3 different targets, these will not always
+   * land at the same time; whilst these Pyres are traveling you can also cast a
+   * new Pyre before the previous ones have landed.
+   * Ontop of this we also have the talent Volatility, which gives your Pyre a
+   * chance to flare up and "recast". These procs can also chain of themselves.
+   *
+   * An example would be you casting Pyre->Dragonrage, you now have 4 active pyres,
+   * each one has a chance to proc Volatility.
+   * In this example we now expect to have 4 seperate sets of damage events in perfect order.
+   * but with differing traveltime and Volatility to take into account we will have a
+   * mix match of events along with a, potentially, infinite extra sets of
+   * damage events. Each individual pyre *should* have all it's damage events in the same tick.
+   *
+   * Due to all of this, it is fairly annoying to attribute damage events to any singular
+   * Pyre.
+   * Solution used is to compare each damage in a given tick and see how much each individual
+   * enemy is hit, if they are hit multiple times we can safely assume it was from
+   * another unique Pyre cast.
```

Comment:
one assumption i might try to add to this is to see if damage events are grouped by pyre instance (rather than being allowed to interleave). e.g. even if every event happens at the same millisecond would the logs look like this?
pyre A hits enemy 1
pyre A hits enemy 2
pyre A hits enemy 3
pyre B hits enemy 2  -- first repeated target, must be a new pyre
pyre B hits enemy 3
pyre B hits enemy 4

if so could simplify this logic a bit. i looked at a few p2 sark pulls and this looks like it would create a decent partitioning.

Reply:
Just so I'm sure I understand this correctly, you are suggesting to simply reset the counter if a repeat target is hit?

Reply:
yeah, or if more than `buffer` time has passed. you'd know better than me if you've seen any data that contradicts this assumption though. i only looked at a very small number of logs to see if it was possible.

Reply:
Tried it out with a couple m+ logs, gives the same results as current iteration, thanks for the input! :)